File: 算法/121. 买卖股票的最佳时机/MaxProfit.go
package _21__买卖股票的最佳时机

func maxProfit(prices []int) int {
	minNumber := prices[0]
	maxNumber := 0
	for i := 1; i < len(prices); i++ {
		maxNumber = max(maxNumber, prices[i]-minNumber)
		minNumber = min(minNumber, prices[i])
	}
	return maxNumber
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

File: 算法/121. 买卖股票的最佳时机/MaxProfit.java

public class MaxProfit {

    class Solution {
        public int maxProfit(int[] prices) {
            // 记录今天之前买入最小值和记录卖出的最大利润，比较每天的最大利润
            int min = prices[0];
            int max = 0;
            for (int i = 1; i < prices.length; i++) {
                max = Math.max(max, prices[i] - min);
                min = Math.min(min, prices[i]);
            }
            return max;
        }
    }

}
